pkg/transformerconfig/factory.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package transformerconfig
+
+import (
+	"log"
+
+	"github.com/ghodss/yaml"
+	"sigs.k8s.io/kustomize/pkg/ifc"
+	"sigs.k8s.io/kustomize/pkg/transformerconfig/defaultconfig"
+)
+
+// Factory makes instances of TransformerConfig.
+type Factory struct {
+	loader ifc.Loader
+}
+
+func NewFactory(l ifc.Loader) *Factory {
+	return &Factory{loader: l}
+}
+
+func (tf *Factory) Loader() ifc.Loader {
+	if tf.loader.(ifc.Loader) == nil {
+		log.Fatal("no loader")
+	}
+	return tf.loader
+}
+
+// FromFiles returns a TranformerConfig object from a list of files
+func (tf *Factory) FromFiles(
+	paths []string) (*TransformerConfig, error) {
+	result := &TransformerConfig{}
+	for _, path := range paths {
+		data, err := tf.Loader().Load(path)
+		if err != nil {
+			return nil, err
+		}
+		t, err := makeTransformerConfigFromBytes(data)
+		if err != nil {
+			return nil, err
+		}
+		result = result.Merge(t)
+	}
+	return result, nil
+}
+
+// makeTransformerConfigFromBytes returns a TransformerConfig object from bytes
+func makeTransformerConfigFromBytes(data []byte) (*TransformerConfig, error) {
+	var t TransformerConfig
+	err := yaml.Unmarshal(data, &t)
+	if err != nil {
+		return nil, err
+	}
+	t.sortFields()
+	return &t, nil
+}
+
+// MakeEmptyTransformerConfig returns an empty TransformerConfig object
+func (tf *Factory) EmptyConfig() *TransformerConfig {
+	return &TransformerConfig{}
+}
+
+// MakeDefaultTransformerConfig returns a default TransformerConfig.
```

Comment:
update this comment
s/MakeDefaultTransformerConfig/DefaultConfig

Reply:
Done
and i have to figure out why the linter didn't catch that